akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentBehaviorImpl.scala

```diff
@@ -122,8 +125,20 @@ private[akka] final case class PersistentBehaviorImpl[Command, Event, State](
 
   /**
    * The `tagger` function should give event tags, which will be used in persistence query
+   *
+   * This is a convenience function that uses withWrapper to wrap the event in Tags.
    */
   def withTagger(tagger: Event ⇒ Set[String]): PersistentBehavior[Command, Event, State] =
-    copy(tagger = tagger)
+    eventWrapper(new TaggingEventWrapper[Event](tagger))
```

Comment:
The we would have to support more than one `EventWrapper` and then the types would go haywire?
Perhaps keep tags separate?

Reply:
:+1:

Reply:
Given that journals don't return the Tagged back it doesn't fit with the Wrapper anyhow